schema/generator.go

```diff
@@ -396,13 +396,26 @@ func (g *Generator) generateDDLsForCreateTable(currentTable Table, desired Creat
 
 		if currentIndex := findIndexByName(currentTable.indexes, desiredIndex.name); currentIndex != nil {
 			// Drop and add index as needed.
-			if !areSameIndexes(*currentIndex, desiredIndex) {
-				ddls = append(ddls, fmt.Sprintf("ALTER TABLE %s DROP INDEX %s", g.escapeTableName(desired.table.name), g.escapeSQLName(currentIndex.name)))
-				ddls = append(ddls, fmt.Sprintf("ALTER TABLE %s ADD %s", g.escapeTableName(desired.table.name), g.generateIndexDefinition(desiredIndex)))
+			switch g.mode {
+			case GeneratorModeMssql:
```

Comment:
This is applicable to all of your patches, but please consider NOT to introduce `g.mode` to a place where `g.mode` switch is not used. At least I couldn't understand why you needed to exclude `areSameIndexOptions` from `areSameIndexes`, and using a name like `generateDDLForCreateIndex` only for mssql is pretty confusing while you already have `generateIndexDefinition` for the same purpose.